darts/timeseries.py

```diff
@@ -2438,94 +2465,134 @@ def with_columns_renamed(
-        return self.pd_dataframe(copy=False).skew(
-            axis, skipna, level, numeric_only, **kwargs
-        )
+        new_xa = xr.DataArray(new_data, dims=self._xa.dims, coords=new_coords)
+        return self.__class__(new_xa)
 
-    def kurtosis(
-        self, axis=None, skipna=True, level=None, numeric_only=None, **kwargs
-    ) -> float:
-        """Simple wrapper around :func:`pd.DataFrame.kurtosis()` for deterministic series."""
-        return self.pd_dataframe(copy=False).kurtosis(
-            axis, skipna, level, numeric_only, **kwargs
+    def median(self, axis: int = 2) -> "TimeSeries":
+        """Simple wrapper around :func:`np.median()`."""
+        new_data = np.median(
+            self._xa.values, axis=axis, overwrite_input=False, keepdims=True
         )
+        new_coords = self._xa.coords
 
-    def min(
-        self, axis=None, skipna=True, level=None, numeric_only=None, **kwargs
-    ) -> float:
-        """Simple wrapper around :func:`pd.DataFrame.min()` for deterministic series."""
-        return self.pd_dataframe(copy=False).min(
-            axis, skipna, level, numeric_only, **kwargs
-        )
+        if axis == 1:  # if we aggregate over components we need to rename components
+            cname = "components_median"
+            new_coords = {self._xa.dims[0]: self.time_index, DIMS[1]: pd.Index([cname])}
 
-    def max(
-        self, axis=None, skipna=True, level=None, numeric_only=None, **kwargs
-    ) -> float:
-        """Simple wrapper around :func:`pd.DataFrame.max()` for deterministic series."""
-        return self.pd_dataframe(copy=False).max(
-            axis, skipna, level, numeric_only, **kwargs
-        )
+        if axis == 0:  # if we aggregate over time we need to change timeindex
+            new_coords = {
+                self._xa.dims[0]: self.time_index[-1:],
+                DIMS[1]: self.components,
+            }
 
-    def sum(
-        self,
-        axis=None,
-        skipna=None,
-        level=None,
-        numeric_only=None,
-        min_count=0,
-        **kwargs,
-    ) -> float:
-        """Simple wrapper around :func:`pd.DataFrame.sum()` for deterministic series."""
-        return self.pd_dataframe(copy=False).sum(
-            axis, skipna, level, numeric_only, min_count, **kwargs
-        )
+        new_xa = xr.DataArray(new_data, dims=self._xa.dims, coords=new_coords)
+        return self.__class__(new_xa)
 
-    def median(
-        self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs
-    ) -> float:
-        """Simple wrapper around :func:`pd.DataFrame.median()` for deterministic series."""
-        return self.pd_dataframe(copy=False).median(
-            axis, skipna, level, numeric_only, **kwargs
-        )
+    def sum(self, axis: int = 2) -> "TimeSeries":
+        """Simple wrapper around :func:`np.ndarray.sum()`."""
+        new_data = self._xa.values.sum(axis=axis, keepdims=True)
```

Comment:
Why not relying on the DataArray functions directly when possible?
E.g.: https://xarray.pydata.org/en/stable/generated/xarray.DataArray.sum.html
Maybe it would do the coordinates management for you when possible.